src/wasm/WasmModule.cpp

```diff
@@ -322,26 +300,6 @@ void WasmModule::writeWasmEnvToMemory(uint32_t envPointers, uint32_t envBuffer)
     throw std::runtime_error("writeWasmEnvToMemory not implemented");
 }
 
-void WasmModule::writeMemoryToFd(int fd)
-{
-    throw std::runtime_error("writeMemoryToFd not implemented");
-}
-
-void WasmModule::mapMemoryFromFd()
-{
-    throw std::runtime_error("mapMemoryFromFd not implemented");
-}
-
-void WasmModule::doSnapshot(std::ostream& outStream)
-{
-    throw std::runtime_error("doSnapshot not implemented");
-}
-
-void WasmModule::doRestore(std::istream& inStream)
-{
-    throw std::runtime_error("doRestore not implemented");
-}
```

Comment:
Submodules of `WasmModule` (i.e. `WAMRWasmModule` and `WAVMWasmModule`) no longer need to provide custom snapshot/ restore behaviour, they just need to be able to provide a pointer to their memory array.